docker-compose.yml

```diff
@@ -17,22 +17,67 @@ services:
       MONGO_DATABASE: ${MONGO_DB_NAME}
       MONGO_USER: ${MONGO_USER}
       MONGO_PASSWORD: ${MONGO_PASSWORD}
-      MONGO_HOST: mongodb
+      MONGO_HOST: mongodb-1
       MONGO_PORT: 27017
+      MONGO_REPL_SET_NAME: ${MONGO_REPL_SET_NAME}
       REDIS_HOST: redis
 
-  mongodb:
+  mongodb-1:
     image: mongo:latest
     restart: always
     volumes:
-      - mongo-data:/data/db
+      - mongo-data1:/data/db
       - ./lib/dump:/docker-entrypoint-initdb.d
+    # - ./db-init/:/docker-entrypoint-initdb.d
+      - ./mongodb-keyfile:/data/mongodb-keyfile
     environment:
       MONGO_INITDB_ROOT_USERNAME: ${MONGO_ROOT_USER}
       MONGO_INITDB_ROOT_PASSWORD: ${MONGO_ROOT_PASSWORD}
       MONGO_INITDB_DATABASE: ${MONGO_DB_NAME}
       MONGO_USER: ${MONGO_USER}
       MONGO_PASSWORD: ${MONGO_PASSWORD}
+    command: --keyFile /data/mongodb-keyfile --replSet ${MONGO_REPL_SET_NAME}
+
+  mongodb-2:
+    image: mongo:latest
+    restart: always
+    volumes:
+      - mongo-data2:/data/db
+      - ./lib/dump:/docker-entrypoint-initdb.d
+      - ./mongodb-keyfile:/data/mongodb-keyfile
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: ${MONGO_ROOT_USER}
+      MONGO_INITDB_ROOT_PASSWORD: ${MONGO_ROOT_PASSWORD}
+      MONGO_INITDB_DATABASE: ${MONGO_DB_NAME}
+      MONGO_USER: ${MONGO_USER}
+      MONGO_PASSWORD: ${MONGO_PASSWORD}
+    command: --keyFile /data/mongodb-keyfile --replSet ${MONGO_REPL_SET_NAME}
+
+  mongodb-3:
+    image: mongo:latest
+    restart: always
+    volumes:
+      - mongo-data3:/data/db
+      - ./lib/dump:/docker-entrypoint-initdb.d
+      - ./mongodb-keyfile:/data/mongodb-keyfile
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: ${MONGO_ROOT_USER}
+      MONGO_INITDB_ROOT_PASSWORD: ${MONGO_ROOT_PASSWORD}
+      MONGO_INITDB_DATABASE: ${MONGO_DB_NAME}
+      MONGO_USER: ${MONGO_USER}
+      MONGO_PASSWORD: ${MONGO_PASSWORD}
+    command: --keyFile /data/mongodb-keyfile --replSet ${MONGO_REPL_SET_NAME}
+
+  mongodb-repl-set-init:
```

Comment:
looks like the mongodb-repli is its own docker image? so there appears to be a leader, rather than just primary and secondary.